src/main/java/duke/parser/Command.java

```diff
@@ -0,0 +1,136 @@
+package duke.parser;
+
+import duke.task.TaskList;
+import duke.task.DeadlineTask;
+import duke.task.EventTask;
+import duke.task.Task;
+import duke.task.ToDoTask;
+import duke.storage.Storage;
+import duke.ui.UiText;
+
+import java.util.Scanner;
+
+import java.util.List;
+
+import java.util.stream.Collectors;
+
+public class Command {
+    Scanner terms;
+
+    Command(Scanner sc) {
+        this.terms = sc;
+    }
+
+    /**
+     * Add new user task into task list
+     *
+     * @param first   = task type specified by user
+     * @param terms   = subsequent terms specified by to define the task
+     * @param tasks   = task list
+     * @param ui      = ui for responding to user after adding the new task
+     * @param storage = save to local file after every changes to task list
+     * @return task successfully added?
+     */
+    static boolean addTask(String first, Scanner terms, TaskList tasks, UiText ui, Storage storage) {

```

Comment:
You can have one child class per specific command as a better way of dealing with multiple commands. This would allow you to make use of polymorphism and code reusability thought your project. There is no need to change the code. But think how you can handle this by considering each specific command as a child class of Command. e.g. AddCommand